api/api-archive-search/archive-search-commons/src/main/java/fr/gouv/vitamui/archives/search/common/dto/converter/JsonPatchToSetActionConverter.java

```diff
@@ -0,0 +1,69 @@
+ * high volumetry securely and efficiently.
+ *
+ * This software is governed by the CeCILL 2.1 license under French law and abiding by the rules of distribution of free
+ * software. You can use, modify and/ or redistribute the software under the terms of the CeCILL 2.1 license as
+ * circulated by CEA, CNRS and INRIA at the following URL "https://cecill.info".
+ *
+ * As a counterpart to the access to the source code and rights to copy, modify and redistribute granted by the license,
+ * users are provided only with a limited warranty and the software's author, the holder of the economic rights, and the
+ * successive licensors have only limited liability.
+ *
+ * In this respect, the user's attention is drawn to the risks associated with loading, using, modifying and/or
+ * developing or reproducing the software by the user in light of its specific status of free software, that may mean
+ * that it is complicated to manipulate, and that also therefore means that it is reserved for developers and
+ * experienced professionals having in-depth computer knowledge. Users are therefore encouraged to load and test the
+ * software's suitability as regards their requirements in conditions enabling the security of their systems and/or data
+ * to be ensured and, more generally, to use and operate it in the same conditions as regards security.
+ *
+ * The fact that you are presently reading this means that you have had knowledge of the CeCILL 2.1 license and that you
+ * accept its terms.
+ */
+
+package fr.gouv.vitamui.archives.search.common.dto.converter;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import fr.gouv.vitam.common.database.builder.query.action.SetAction;
+import fr.gouv.vitam.common.database.builder.query.action.UpdateActionHelper;
+import fr.gouv.vitam.common.database.builder.request.exception.InvalidCreateOperationException;
+import fr.gouv.vitamui.archives.search.common.model.JsonPatch;
+import fr.gouv.vitamui.archives.search.common.model.PatchCommand;
+import fr.gouv.vitamui.archives.search.common.model.PatchOperation;
+import fr.gouv.vitamui.commons.api.logger.VitamUILogger;
+import fr.gouv.vitamui.commons.api.logger.VitamUILoggerFactory;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.stereotype.Service;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Service
+public class JsonPatchToSetActionConverter implements Converter<JsonPatch, SetAction> {
+
+    private static final VitamUILogger log =
+        VitamUILoggerFactory.getInstance(JsonPatchToSetActionConverter.class);
+
+    @Override
+    public SetAction convert(JsonPatch source) {
+        final Map<String, JsonNode> map = source.stream()
+            .filter(patchCommand -> {
+                final boolean isAdd = patchCommand.getOp() == PatchOperation.ADD;
+                final boolean isReplace = patchCommand.getOp() == PatchOperation.REPLACE;
+                return isAdd || isReplace;
+            })
```

Comment:
Pourrait être remplacé par :
```suggestion
.filter(patchCommand -> Arrays.asList(PatchOperation.ADD, PatchOperation.REPLACE).contains(patchCommand.getOp()))
```

Reply:
J'ai remplacé par l'équivalence List.of